homeassistant/components/snooz/config_flow.py

```diff
@@ -31,11 +35,16 @@ class DiscoveredSnooz:
     info: BluetoothServiceInfo
     device: SnoozAdvertisementData
 
+    @property
+    def display_name(self) -> str:
+        """Return a display name for the device."""
+        return get_device_display_name(self.info.name, self.info.address)
+
 
 class SnoozConfigFlow(ConfigFlow, domain=DOMAIN):
     """Handle a config flow for Snooz."""
 
-    VERSION = 1
+    VERSION = 2
```

Comment:
You updated the version. This means you need a migration. I only glanced at the code so far (am traveling so I'm on my phone so I can't get a full picture) and what I understood is that we now store more options. We can also do that without a migration by adding the required information in the init (although I am currently reworking it, checkout the current withings integration. In `__init__.py` at the `async_setup_entry` extra info is gathered to update the config entry. Would this work?)

The downside of doing a migration is that it's not reversable. So if this gets in 2023.10, they can't just install 2023.9 and expect it to work, since the config entry has a different version. So although we have the whole migration session, we try to avoid it as much as possible, if it's not needed.

Reply:
Oh I now see you change from pairing token to password? So then a migration can become useful

Reply:
pysnooz changed naming from pairing token to password to be consistent with Snooz's internal naming. In the snooz component, config data will remain `token` since it's just a semantic change.

Pre `0.9.0` only SNOOZ Original was supported. Model detection requires the advertised name with the specific format `Model-XXXX`. I see what you're saying about doing the migration logic. I will include it as part of the setup process instead of storing `model` and `password`.

Is it acceptable for `async_setup_entry` to modify data? One downside to moving the migration logic into `async_setup` is that it will increase setup time by default since often times cached advertisements use the device characteristic name instead of the name from an original advertisement. This is always `Snooz`, so it breaks model detection that requires `Mpdel-XXXX`. We can avoid the startup time by caching `model`.